In [1]:
!conda install -yc conda-forge ipython-sql

Solving environment: ...working... done

# All requested packages already installed.



# Sequel to Data Analyst carrer:
## Exploring global territorial dataset with SQL

![banner](https://www.infoplease.com/sites/infoplease.com/files/styles/webp_image/public/inline-images/Global%20Population.jpg.webp)

## Abstract

In this first contact with SQL, we lay over general global territorial statistics from the CIA to inquire about the outliers on multiple variables. We discover, among others, that India is the fastest growing country, Macau is the most densely populated, and British Indian Ocean Territory is nearly 99.9% covered by water. 

## Table of Contents

1. [**Introduction**](#section1)
2. [**Goals and limitations**](#section2)
3. [**Methodology**](#section3)
4. [**Metadata**](#section4)
5. [**Data Analysis**](#section5)      
6. [**Conclusion**](#section6)

## 1. Introduction<a name="section1"></a>


This project deals with the [CIA World Factbook](https://www.cia.gov/the-world-factbook/), where general statistics about all countries on Earth can be found. We'll use SQL in Jupyter to find generalizations about the countries.

## 2. Goals and limitations<a name="section2"></a>

In this project, we investigate:

* (1) which countries have the largest and smallest populations in the dataset;
* (2) the lowest population growth rate;
* (3) the most densely populated countries;
* (4) which countries have the highest ration of water to land, and more water than land?
* (5) Which countries will add the most people to their populations next year?
* (6) Which countries have higher death rate than birth rate?
* (7) Which countries have the highest population/area ratio, and how does it compare to list we found in the previous screen?

Limitations:
* as our first project with SQL, it is quite expected for the data manipulations to be simpler.

## 3. Methodology<a name="section3"></a>

We'll explore the dataset with basic concepts of SQLite: SELECT FROM, WHERE, LIMIT, NOT IN, AND, ORDER BY, CAST.

## 4. Metadata<a name="section4"></a>

Here are the descriptions for some of the columns:

* name — the name of the country.
* area— the country's total area (both land and water).
* area_land — the country's land area in square kilometers.
* area_water — the country's water area in square kilometers.
* population — the country's population.
* population_growth— the country's population growth as a percentage.
* birth_rate — the country's birth rate, or the number of births per year per 1,000 people.
* death_rate — the country's death rate, or the number of death per year per 1,000 people.

## 5. Data Analysis<a name="section5"></a>

In this project to familiarize us with SQL, we assume the data has no mistakes.

In [2]:
#%%capture
%load_ext sql
%sql sqlite:///factbook.db
    


'Connected: @factbook.db'

In [4]:
%%sql
SELECT *
FROM sqlite_master
WHERE type='table'

 * sqlite:///factbook.db
Done.


[('table', 'sqlite_sequence', 'sqlite_sequence', 3, 'CREATE TABLE sqlite_sequence(name,seq)'),
 ('table', 'facts', 'facts', 47, 'CREATE TABLE "facts" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "code" varchar(255) NOT NULL, "name" varchar(255) NOT NULL, "area" integer, "a ... (4 characters truncated) ... land" integer, "area_water" integer, "population" integer, "population_growth" float, "birth_rate" float, "death_rate" float, "migration_rate" float)')]

In [5]:
%%sql
SELECT *
    FROM facts
LIMIT 5;

 * sqlite:///factbook.db
Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

Write a single querry: min population, max population, minimum population growth, max population growth

In [6]:
%%sql
SELECT MIN(population) AS min_pop, MAX(population) AS max_pop, MIN(population_growth) AS min_pop_growth, MAX(population_growth) AS max_population_growth
    FROM facts;

 * sqlite:///factbook.db
Done.


[(0, 7256490011, 0.0, 4.02)]

* It is proposterus to think a country would have a population of 0;
* How could a country have the population of the whole planet, at 7.25 billion people?

In [10]:
%%sql
SELECT *
    FROM facts
    WHERE population == (SELECT MIN(population)
                           FROM facts);

 * sqlite:///factbook.db
Done.


[(250, 'ay', 'Antarctica', None, 280000, None, 0, None, None, None, None)]

So the 'country' with 0 folks registered in it is Antartica, which is not really a country. This agrees with the CIA Factbook [page for Antartica](https://www.cia.gov/library/publications/the-world-factbook/geos/ay.html).

![banner](https://s3.amazonaws.com/dq-content/257/fb_antarctica.png)

In [9]:
%%sql
SELECT *
    FROM facts
    WHERE population==(SELECT MAX(population)
                         FROM facts);

 * sqlite:///factbook.db
Done.


[(261, 'xx', 'World', None, None, None, 7256490011, 1.08, 18.6, 7.8, None)]

As we could foresee, there is an entry for 'World' with all living population in it. As both minimum and maximum are not satisfactory, we should recalculate excluding these statistics.

In [12]:
%%sql
SELECT MIN(population) AS min_pop, MAX(population) AS max_pop, MIN(population_growth) AS min_pop_growth, MAX(population_growth) AS max_population_growth
    FROM facts
    WHERE name NOT IN ('World', 'Antarctica');

 * sqlite:///factbook.db
Done.


[(48, 1367485388, 0.0, 4.02)]

In [15]:
%%sql
SELECT AVG(population) AS avg_pop, AVG(area) AS avg_area
    FROM facts
    WHERE name NOT IN ('World', 'Antarctica');

 * sqlite:///factbook.db
Done.


[(32377011.0125, 555093.546184739)]

So the average population in the dataset is 32 million citizens per country, averaging 555 thousand square kilometers.

* (1) which countries have the largest and smallest populations in the dataset?

In [25]:
%%sql
SELECT *
    FROM facts
    WHERE name NOT IN ('Antarctica', 'World')
    AND population == (SELECT MAX(population)
                          FROM facts
                          WHERE name NOT IN ('Antarctica', 'World'))

 * sqlite:///factbook.db
Done.


[(37, 'ch', 'China', 9596960, 9326410, 270550, 1367485388, 0.45, 12.49, 7.53, 0.44)]

In [26]:
%%sql
SELECT *
    FROM facts
    WHERE name NOT IN ('Antarctica', 'World')
    AND population == (SELECT MIN(population)
                          FROM facts
                          WHERE name NOT IN ('Antarctica', 'World'))

 * sqlite:///factbook.db
Done.


[(238, 'pc', 'Pitcairn Islands', 47, 47, 0, 48, 0.0, None, None, None)]

The countries with the largest and smallest populations are China and Pitcairn Islands, respectively.

* (2) the lowest population growth rate?

In [32]:
%%sql
SELECT *
    FROM facts
    WHERE (name NOT IN ('Antarctica','World'))
    AND (population_growth == (SELECT MIN(population_growth)
            FROM facts));

 * sqlite:///factbook.db
Done.


[(190, 'vt', 'Holy See (Vatican City)', 0, 0, 0, 842, 0.0, None, None, None),
 (200, 'ck', 'Cocos (Keeling) Islands', 14, 14, 0, 596, 0.0, None, None, None),
 (207, 'gl', 'Greenland', 2166086, 2166086, None, 57733, 0.0, 14.48, 8.49, 5.98),
 (238, 'pc', 'Pitcairn Islands', 47, 47, 0, 48, 0.0, None, None, None)]

Populations with the lowest growth rate are: Greenland, Vatican, Cocos (Kelling) Islands, Pitcairn Islands.

* (3) the largest densely populated countries?

In [49]:
%%sql
SELECT name AS country
    FROM facts
    WHERE name NOT IN ('WORLD', 'ANTARCTICA') AND (population > (SELECT AVG(population)
                           FROM facts
                           WHERE name NOT IN ('World', 'Antarctica'))) AND (area < (SELECT AVG(area)
                           FROM facts
                           WHERE name NOT IN ('World', 'Antarctica')))
    ORDER BY area DESC;

 * sqlite:///factbook.db
Done.


[('Thailand',),
 ('Spain',),
 ('Morocco',),
 ('Iraq',),
 ('Japan',),
 ('Germany',),
 ('Vietnam',),
 ('Poland',),
 ('Italy',),
 ('Philippines',),
 ('United Kingdom',),
 ('Uganda',),
 ('Bangladesh',),
 ('Korea, South',)]

The largest densely populated countries are: Thailand, Spain, Morrocco, Iraq, Japan;

* (4) which countries have the highest ration of water to land, and more water than land? 

In [42]:
%%sql
SELECT *, CAST(area_water AS FLOAT)/area_land AS water_land_ratio
    FROM facts
    ORDER BY water_land_ratio DESC
    LIMIT 5;

 * sqlite:///factbook.db
Done.


[(228, 'io', 'British Indian Ocean Territory', 54400, 60, 54340, None, None, None, None, None, 905.6666666666666),
 (247, 'vq', 'Virgin Islands', 1910, 346, 1564, 103574, 0.59, 10.31, 8.54, 7.67, 4.520231213872832),
 (246, 'rq', 'Puerto Rico', 13791, 8870, 4921, 3598357, 0.6, 10.86, 8.67, 8.15, 0.5547914317925592),
 (12, 'bf', 'Bahamas, The', 13880, 10010, 3870, 324597, 0.85, 15.5, 7.05, 0.0, 0.3866133866133866),
 (71, 'pu', 'Guinea-Bissau', 36125, 28120, 8005, 1726170, 1.91, 33.38, 14.33, 0.0, 0.2846728307254623)]

The countries with more water than land are the British Indian Ocean Territory at 905 times water area, and the Virgin Islands, with 4.52 times the land area in water.

* (5) Which countries will add the most people to their populations next year ?

In [44]:
%%sql
SELECT name, population*population_growth AS new_folks
    FROM facts
    ORDER BY new_folks DESC;

 * sqlite:///factbook.db
Done.


[('World', 7837009211.88),
 ('India', 1527068612.48),
 ('China', 615368424.6),
 ('Nigeria', 444827037.20000005),
 ('Pakistan', 290665336.62),
 ('Ethiopia', 287456216.91),
 ('Bangladesh', 270332392.0),
 ('United States', 250667713.92000002),
 ('Indonesia', 235514180.08),
 ('Congo, Democratic Republic of the', 194469083.20000002),
 ('Philippines', 162607385.36),
 ('Egypt', 158392438.84),
 ('Brazil', 157280055.24),
 ('Mexico', 143649434.62),
 ('Tanzania', 142418010.78),
 ('European Union', 128487361.25),
 ('Uganda', 120209653.80000001),
 ('Iraq', 108574575.17),
 ('Turkey', 100061978.94),
 ('Iran', 98189124.0),
 ('Vietnam', 91518369.95),
 ('Kenya', 88635830.92999999),
 ('Afghanistan', 75549273.44),
 ('Algeria', 72757585.44),
 ('South Africa', 71388498.79),
 ('Yemen', 66041172.99),
 ('Sudan', 62107227.16),
 ('Mozambique', 61992626.85),
 ('Cameroon', 61484574.62),
 ('Madagascar', 61436716.980000004),
 ('Malawi', 59642794.04),
 ('Niger', 58648619.25),
 ('Ghana', 57394274.82000001),
 ('Burkina Faso', 57363008.58),
 ('Burma', 56883408.06),
 ('Nepal', 56476835.95),
 ('Angola', 54558481.339999996),
 ('Mali', 50527497.28),
 ('Colombia', 48606197.120000005),
 ('South Sudan', 48412498.199999996),
 ("Cote d'Ivoire", 44494026.82),
 ('Malaysia', 43939941.12),
 ('Zambia', 43390846.08),
 ('Spain', 42850059.26),
 ('Venezuela', 40692889.4),
 ('Saudi Arabia', 40518381.36),
 ('Argentina', 40391653.980000004),
 ('Burundi', 35234665.28),
 ('United Kingdom', 34607639.88),
 ('Senegal', 34240793.300000004),
 ('Morocco', 33322699.0),
 ('Rwanda', 32414036.48),
 ('Zimbabwe', 31447285.61),
 ('Guinea', 30981826.06),
 ('Peru', 29531649.029999997),
 ('Benin', 29047238.659999996),
 ('France', 28618119.38),
 ('Uzbekistan', 27155946.060000002),
 ('Guatemala', 27152578.18),
 ('Ukraine', 26657682.599999998),
 ('Canada', 26324877.0),
 ('Cambodia', 24819834.48),
 ('Australia', 24343584.98),
 ('Thailand', 23111977.700000003),
 ('Chad', 21983451.84),
 ('Ecuador', 21422334.6),
 ('Kazakhstan', 20699119.08),
 ('Togo', 20315735.419999998),
 ('Japan', 20307145.44),
 ('Somalia', 19427975.400000002),
 ('Sri Lanka', 18524929.919999998),
 ('Bolivia', 16849375.92),
 ('Italy', 16700882.4),
 ('United Arab Emirates', 14911780.8),
 ('Honduras', 14694410.639999999),
 ('Eritrea', 14687300.25),
 ('Chile', 14356773.2),
 ('Libya', 14298260.48),
 ('Tajikistan', 14008248.18),
 ('Sierra Leone', 13815880.3),
 ('Germany', 13745249.360000001),
 ('Korea, North', 13241098.65),
 ('Dominican Republic', 12888869.879999999),
 ('Israel', 12556929.84),
 ('Papua New Guinea', 11876923.620000001),
 ('Haiti', 11828722.229999999),
 ('Central African Republic', 11483978.07),
 ('Singapore', 10724752.08),
 ('Laos', 10712893.200000001),
 ('Liberia', 10363295.020000001),
 ('Tunisia', 9823130.25),
 ('Congo, Republic of the', 9510194.0),
 ('Azerbaijan', 9389548.799999999),
 ('Belgium', 8606219.48),
 ('Mauritania', 8020645.46),
 ('Paraguay', 7868595.52),
 ('Sweden', 7841292.800000001),
 ('Netherlands', 6948640.64),
 ('Korea, South', 6876127.44),
 ('Oman', 6803957.52),
 ('Qatar', 6738088.1899999995),
 ('Jordan', 6737578.12),
 ('Romania', 6499905.0),
 ('Kyrgyzstan', 6288082.290000001),
 ('Ireland', 6115381.25),
 ('Turkmenistan', 5963821.079999999),
 ('Nicaragua', 5907881.0),
 ('Norway', 5884688.569999999),
 ('Costa Rica', 5873255.68),
 ('Switzerland', 5766499.3),
 ('Russia', 5696950.92),
 ('West Bank', 5431463.7),
 ('Taiwan', 5385478.98),
 ('Lebanon', 5318842.86),
 ('Gaza Strip', 5252044.55),
 ('Panama', 4827271.680000001),
 ('Austria', 4766052.5),
 ('Kuwait', 4517425.08),
 ('Gambia, The', 4250251.44),
 ('Bulgaria', 4168397.94),
 ('Mongolia', 3920709.48),
 ('Moldova', 3653252.41),
 ('New Zealand', 3639482.26),
 ('Poland', 3470597.01),
 ('Serbia', 3301325.24),
 ('Guinea-Bissau', 3296984.6999999997),
 ('Gabon', 3291298.48),
 ('Bahrain', 3245337.33),
 ('Lithuania', 2999810.3200000003),
 ('Timor-Leste', 2979300.7199999997),
 ('Syria', 2730376.64),
 ('Hong Kong', 2713620.2800000003),
 ('Botswana', 2641089.9899999998),
 ('

India is expected to more than double its size next year, with 1.53 billion new borns. China ranks second with 600 million new children.

* (6) Which countries have higher death rate than birth rate?

In [45]:
%%sql
SELECT *, birth_rate - death_rate AS net_rate
    FROM facts
    WHERE net_rate<0
    ORDER BY net_rate
    ;

 * sqlite:///factbook.db
Done.


[(26, 'bu', 'Bulgaria', 110879, 108489, 2390, 7186893, 0.58, 8.92, 14.44, 0.29, -5.52),
 (153, 'ri', 'Serbia', 77474, 77474, 0, 7176794, 0.46, 9.08, 13.66, 0.0, -4.58),
 (96, 'lg', 'Latvia', 64589, 62249, 2340, 1986705, 1.06, 10.0, 14.31, 6.26, -4.3100000000000005),
 (102, 'lh', 'Lithuania', 65300, 62680, 2620, 2884433, 1.04, 10.1, 14.27, 6.27, -4.17),
 (183, 'up', 'Ukraine', 603550, 579330, 24220, 44429471, 0.6, 10.72, 14.46, 2.25, -3.74),
 (75, 'hu', 'Hungary', 93028, 89608, 3420, 9897541, 0.22, 9.16, 12.73, 1.33, -3.5700000000000003),
 (65, 'gm', 'Germany', 357022, 348672, 8350, 80854408, 0.17, 8.47, 11.42, 1.24, -2.9499999999999993),
 (158, 'si', 'Slovenia', 20273, 20151, 122, 1983412, 0.26, 8.42, 11.37, 0.37, -2.9499999999999993),
 (142, 'ro', 'Romania', 238391, 229891, 8500, 21666350, 0.3, 9.14, 11.9, 0.24, -2.76),
 (44, 'hr', 'Croatia', 56594, 55974, 620, 4464844, 0.13, 9.45, 12.18, 1.39, -2.7300000000000004),
 (16, 'bo', 'Belarus', 207600, 202900, 4700, 9589689, 0.2, 10.7, 13.36, 0.7, -2.66),
 (117, 'mn', 'Monaco', 2, 2, 0, 30535, 0.12, 6.65, 9.24, 3.83, -2.59),
 (67, 'gr', 'Greece', 131957, 130647, 1310, 10775643, 0.01, 8.66, 11.09, 2.32, -2.4299999999999997),
 (214, 'sb', 'Saint Pierre and Miquelon', 242, 242, 0, 5657, 1.08, 7.42, 9.72, 8.49, -2.3000000000000007),
 (143, 'rs', 'Russia', 17098242, 16377742, 720500, 142423773, 0.04, 11.6, 13.69, 1.69, -2.09),
 (57, 'en', 'Estonia', 45228, 42388, 2840, 1265420, 0.55, 10.51, 12.4, 3.6, -1.8900000000000006),
 (140, 'po', 'Portugal', 92090, 91470, 620, 10825309, 0.09, 9.27, 11.02, 2.67, -1.75),
 (85, 'ja', 'Japan', 377915, 364485, 13430, 126919659, 0.16, 7.93, 9.51, 0.0, -1.58),
 (83, 'it', 'Italy', 301340, 294140, 7200, 61855120, 0.27, 8.74, 10.19, 4.1, -1.4499999999999993),
 (22, 'bk', 'Bosnia and Herzegovina', 51197, 51187, 10, 3867055, 0.13, 8.87, 9.75, 0.38, -0.8800000000000008),
 (47, 'ez', 'Czech Republic', 78867, 77247, 1620, 10644842, 0.16, 9.63, 10.34, 2.33, -0.7099999999999991),
 (116, 'md', 'Moldova', 33851, 32891, 960, 3546847, 1.03, 12.0, 12.59, 9.67, -0.5899999999999999),
 (139, 'pl', 'Poland', 312685, 304255, 8430, 38562189, 0.09, 9.74, 10.19, 0.46, -0.4499999999999993),
 (10, 'au', 'Austria', 83871, 82445, 1426, 8665550, 0.55, 9.41, 9.42, 5.56, -0.009999999999999787)]

There is a long list of countries shrinking their populations, headed by Bulgaria (-5,5%), Serbia, Latvia, Lithuania above 4%.

* (7) Which countries have the highest population/area ratio, and how does it compare to list we found in the previous screen?

In [47]:
%%sql
SELECT *, population/area AS pop_density
    FROM facts
    ORDER BY pop_density DESC;    

 * sqlite:///factbook.db
Done.


[(205, 'mc', 'Macau', 28, 28, 0, 592731, 0.8, 8.88, 4.22, 3.37, 21168),
 (117, 'mn', 'Monaco', 2, 2, 0, 30535, 0.12, 6.65, 9.24, 3.83, 15267),
 (156, 'sn', 'Singapore', 697, 687, 10, 5674472, 1.89, 8.27, 3.43, 14.05, 8141),
 (204, 'hk', 'Hong Kong', 1108, 1073, 35, 7141106, 0.38, 9.23, 7.07, 1.68, 6445),
 (251, 'gz', 'Gaza Strip', 360, 360, 0, 1869055, 2.81, 31.11, 3.04, 0.0, 5191),
 (233, 'gi', 'Gibraltar', 6, 6, 0, 29258, 0.24, 14.08, 8.37, 3.28, 4876),
 (13, 'ba', 'Bahrain', 760, 760, 0, 1346613, 2.41, 13.66, 2.69, 13.09, 1771),
 (108, 'mv', 'Maldives', 298, 298, 0, 393253, 0.08, 15.75, 3.89, 12.68, 1319),
 (110, 'mt', 'Malta', 316, 316, 0, 413965, 0.31, 10.18, 9.09, 1.98, 1310),
 (227, 'bd', 'Bermuda', 54, 54, 0, 70196, 0.5, 11.33, 8.23, 1.88, 1299),
 (218, 'sk', 'Sint Maarten', 34, 34, 0, 39689, 1.51, 13.0, 4.51, 6.63, 1167),
 (14, 'bg', 'Bangladesh', 148460, 130170, 18290, 168957745, 1.6, 21.14, 5.61, 0.46, 1138),
 (234, 'gk', 'Guernsey', 78, 78, 0, 66080, 0.34, 9.84, 8.78, 2.32, 847),
 (235, 'je', 'Jersey', 116, 116, 0, 97294, 0.8, 11.91, 7.68, 3.76, 838),
 (15, 'bb', 'Barbados', 430, 430, 0, 290604, 0.31, 11.87, 8.44, 0.3, 675),
 (113, 'mp', 'Mauritius', 2040, 2030, 10, 1339827, 0.64, 13.29, 6.91, 0.0, 656),
 (196, 'tw', 'Taiwan', 35980, 32260, 3720, 23415126, 0.23, 8.47, 7.11, 0.89, 650),
 (216, 'aa', 'Aruba', 180, 180, 0, 112162, 1.33, 12.56, 8.18, 8.92, 623),
 (97, 'le', 'Lebanon', 10400, 10230, 170, 6184701, 0.86, 14.59, 4.88, 1.1, 594),
 (213, 'rn', 'Saint Martin', 54, 54, None, 31754, None, None, None, None, 588),
 (149, 'sm', 'San Marino', 61, 61, 0, 33020, 0.82, 8.63, 8.45, 8.03, 541),
 (91, 'ks', 'Korea, South', 99720, 96920, 2800, 49115196, 0.14, 8.19, 6.75, 0.0, 492),
 (144, 'rw', 'Rwanda', 26338, 24668, 1670, 12661733, 2.56, 33.75, 8.96, 0.85, 480),
 (254, 'we', 'West Bank', 5860, 5640, 220, 2785366, 1.95, 22.99, 3.5, 0.0, 475),
 (123, 'nr', 'Nauru', 21, 21, 0, 9540, 0.55, 24.95, 5.87, 13.63, 454),
 (181, 'tv', 'Tuvalu', 26, 26, 0, 10869, 0.82, 23.74, 8.74, 6.81, 418),
 (125, 'nl', 'Netherlands', 41543, 33893, 7650, 16947904, 0.41, 10.83, 8.66, 1.95, 407),
 (111, 'rm', 'Marshall Islands', 181, 181, 0, 72191, 1.66, 25.6, 4.21, 4.83, 398),
 (82, 'is', 'Israel', 20770, 20330, 440, 8049314, 1.56, 18.48, 5.15, 2.24, 387),
 (29, 'by', 'Burundi', 27830, 25680, 2150, 10742276, 3.28, 42.01, 9.27, 0.0, 385),
 (77, 'in', 'India', 3287263, 2973193, 314070, 1251695584, 1.22, 19.55, 7.32, 0.04, 380),
 (17, 'be', 'Belgium', 30528, 30278, 250, 11323973, 0.76, 11.41, 9.63, 5.87, 370),
 (73, 'ha', 'Haiti', 27750, 27560, 190, 10110019, 1.17, 22.31, 7.83, 2.76, 364),
 (39, 'cn', 'Comoros', 2235, 2235, 0, 780971, 1.77, 27.84, 7.57, 2.53, 349),
 (138, 'rp', 'Philippines', 300000, 298170, 1830, 100998376, 1.61, 24.27, 6.11, 2.09, 336),
 (164, 'ce', 'Sri Lanka', 65610, 64630, 980, 22053488, 0.84, 15.85, 6.11, 1.35, 336),
 (85, 'ja', 'Japan', 377915, 364485, 13430, 126919659, 0.16, 7.93, 9.51, 0.0, 335),
 (217, 'cc', 'Curacao', 444, 444, 0, 148406, 0.43, 13.8, 8.2, 1.4, 334),
 (68, 'gj', 'Grenada', 344, 344, 0, 110694, 0.48, 16.03, 8.08, 3.13, 321),
 (243, 'gq', 'Guam', 544, 544, 0, 161785, 0.54, 16.82, 5.12, 6.34, 297),
 (54, 'es', 'El Salvador', 21041, 20721, 320, 6141350, 0.25, 16.46, 5.69, 8.28, 291),
 (192, 'vm', 'Vietnam', 331210, 310070, 21140, 94348835, 0.97, 15.96, 5.93, 0.3, 284),
 (242, 'aq', 'American Samoa', 199, 199, 0, 54343, 0.3, 22.89, 4.75, 21.13, 273),
 (84, 'jm', 'Jamaica', 10991, 10831, 160, 2950210, 0.68, 18.16, 6.7, 4.66, 268),
 (146, 'st', 'Saint Lucia', 616, 606, 10, 163922, 0.34, 13.7, 7.42, 2.93, 266),
 (147, 'vc', 'Saint Vincent and the Grenadines', 389, 389, 0, 102627, 0.28, 13.57, 7.18, 9.17, 263),
 (185, 'uk', 'United Kingdom', 243610, 241930, 1680, 64088222, 0.54, 12.17, 9.35, 2.54, 263),
 (246, 'rq', 'Puerto Rico', 13791, 8870, 4921, 3598357, 0.6, 10.86, 8.67, 8.15, 260),
 (132, 'pk', 'Pakistan', 796095, 770875, 25220, 199085847, 1.46, 22.58, 6.49, 1.54, 250),
 (177, 'td', 'Trinidad and Tobago', 512

We can see that the most dense countries are Macau, Monaco, Singapore, Honk Kong and the Gaza Strip, being very different from the previous list where we combined only populations above average with areas bellow average.

## 6. Conclusion<a name="section6"></a>

In this first contact with SQL, we obtained for our queries:

* (1) countries with largest and smallest populations: China, Pitcairn Islands;
* (2) the lowest population growth rate: Greenland, Vatican, Cocos (Kelling) Islands, Pitcairn Islands at 0;
* (3) the largest densely populated countries: Thailand, Spain, Morrocco, Iraq, Japan;
* (4) countries with more water than land: British Indian Ocean Territory, and the Virgin Islands
* (5) countries that will grow the most next year: India and China, at 1.52B and 600M.
* (6) countries with worst growth rates: Bulgaria(-5,5%), Serbia, Latvia, Lithuania (all >4%).
* (7) most densely populated countries overall: Macau, Monaco, Singapore, Honk Kong, Gaza Strip. 